### DISCUSSION TITLE
No sound on Linux Guest using Apple Virtualization

### DISCUSSION THREAD

**Discussion by raonetwo** [utmapp/UTM#5183] [UTM Help (macOS)]
Installed Arch following [this](https://github.com/3rfaan/dotfiles) guide and installed gnome thereafter. 

MacOS Version: 13.3.1 (22E261)  
Processor: M1
UTM Version: 4.1.6

I had selected apple virtualisation, no sound device is being shown in gnome settings.

```
arch# lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 05ac:8106 Apple, Inc. Virtual USB Digitizer
Bus 001 Device 002: ID 05ac:8105 Apple, Inc. Virtual USB Keyboard
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

It appears it is supported by apple https://developer.apple.com/documentation/virtualization/audio 

```
arch# lsmod | grep virtio                            
virtio_gpu             57344  1
virtio_net             61440  0
net_failover           20480  1 virtio_net
virtio_dma_buf         16384  1 virtio_gpu
drm_shmem_helper       24576  1 virtio_gpu
virtio_balloon         24576  0
virtiofs               28672  0
fuse                  139264  6 virtiofs
virtio_rng             16384  0

```

```
arch# zgrep VIRTIO /proc/config.gz
CONFIG_BLK_MQ_VIRTIO=y
CONFIG_VIRTIO_VSOCKETS=m
CONFIG_VIRTIO_VSOCKETS_COMMON=m
# CONFIG_BT_VIRTIO is not set
CONFIG_NET_9P_VIRTIO=m
# CONFIG_ARM_SCMI_TRANSPORT_VIRTIO is not set
CONFIG_VIRTIO_BLK=y
CONFIG_SCSI_VIRTIO=y
CONFIG_VIRTIO_NET=m
CONFIG_VIRTIO_CONSOLE=y
CONFIG_HW_RANDOM_VIRTIO=m
# CONFIG_I2C_VIRTIO is not set
# CONFIG_GPIO_VIRTIO is not set
CONFIG_DRM_VIRTIO_GPU=m
# CONFIG_SND_VIRTIO is not set
CONFIG_VIRTIO_ANCHOR=y
CONFIG_VIRTIO=y
CONFIG_VIRTIO_PCI_LIB=y
CONFIG_VIRTIO_PCI_LIB_LEGACY=y
CONFIG_VIRTIO_MENU=y
CONFIG_VIRTIO_PCI=y
CONFIG_VIRTIO_PCI_LEGACY=y
# CONFIG_VIRTIO_PMEM is not set
CONFIG_VIRTIO_BALLOON=m
CONFIG_VIRTIO_INPUT=m
CONFIG_VIRTIO_MMIO=m
# CONFIG_VIRTIO_MMIO_CMDLINE_DEVICES is not set
CONFIG_VIRTIO_DMA_SHARED_BUFFER=m
# CONFIG_VIRTIO_IOMMU is not set
# CONFIG_RPMSG_VIRTIO is not set
CONFIG_VIRTIO_FS=m
CONFIG_CRYPTO_DEV_VIRTIO=m
```
1. My guess is I will have to enable CONFIG_SND_VIRTIO for this to work, but i am not sure how. It appears I will have to compile a modified kernel. Is there a workaround for this?
2. Alternatively, is there a way to convert this vm to qemu vm?
3. Is there any other way to create a virtual sound device, such that audio from it can then be forwarded to host?

**Comment by acheronfail** [comment]
> My guess is I will have to enable CONFIG_SND_VIRTIO for this to work, but i am not sure how. It appears I will have to compile a modified kernel. 

I cloned https://github.com/archlinuxarm/PKGBUILDs and rebuilt the [PKGBUILD from `linux-aarch64`](https://github.com/archlinuxarm/PKGBUILDs/tree/master/core/linux-aarch64) with `CONFIG_SND_VIRTIO=m` and can confirm sound works fine in an Arch VM using Apple Virtualisation.

You can see the changes I made here: https://github.com/archlinuxarm/PKGBUILDs/commit/fae13fc75ab3fd9699d0e1817917d523d8c8e7c4

**Reply by acheronfail** [reply]
I wrote up a more complete guide on how I setup my Arch Linux VM here with UTM: https://acheronfail.github.io/posts/arch-vm-apple-arm64/

**Comment by anthonywong** [comment]
> I wrote up a more complete guide on how I setup my Arch Linux VM here with UTM: https://acheronfail.github.io/guides/arch-vm-apple-arm64/

The link is no longer accessible, do you have the latest URL?

**Reply by acheronfail** [reply]
Updated 👍
